### DISCUSSION TITLE
using of expm_lanczos:  inconsistent results got when using different sizes of time steps

### DISCUSSION THREAD

**Discussion by chiwhalee** [QuSpin/QuSpin#562] [Q&A]
I want to evolve a state under a Hamiltonian up to t_final~10^12.  I found expm_lanczos is much faster than .evolve(), so I choose to use that  method. I divide [t_initial=0, t_final] by a time interval delta_t, evolve the state and measure at each intermediate time steps.  However,  I found the results are very different  for using different sizes of time steps, say delta_t=100 and delta_t=10000.   I wonder is this normal and expected?  Why are the results  be different.

**Comment by weinbe58** [comment]
Hi,

The larger the step size the larger the number of pancake vectors are required for it to converge. There are two concerns here. 

You might be able to use lanczos to do time steps up to 100 but that's probably it since you have to worry about loss of orthogonality. At a certain point it becomes more efficient to fully disgonalize the Hamiltonian. That is what I would recommend here.

Secondly, practically speaking you can't get to those time scales due to rounding errors. With double precision the rounding error is order 10^-15 so you see the problem with going to times order 10^16

**Reply by chiwhalee** [reply]
Thank you very much for your reply. I want to first make sure whether expm_lanczos  is the most suitable algorithm to solve the problem in the Quspin package?  My goal is to evolve a pure state under a spin-1/2 Hamiltonian, with chain lengths up to 20,  using the symmetry of fixed number of up spins,  total  time up to about 10^12, measuring the magnetization at each time steps (several tens of steps are enough). I need the error for the  magnetization value to be under 0.1 ?  Is this goal practical by using expm_lanczos?   (I saw this kind of simulation in some published papers, but I don't know exactly  which algorithm they have used)

What do you mean by "fully disgonalize the Hamiltonian"?  I had been always setting  the full_ortho to be True in lanczos_full(H, v0, m, full_ortho=True), do you mean that?  If 
 not,  what changes should I make to these lines 
>>> E, V, Q_T = lanczos_full(H,v0, m, full_ortho=True)
>>> expH_v0 = expm_lanczos(E,V,Q_T,a=-1j)
